encfs/FileUtils.cpp

```diff
@@ -644,7 +644,7 @@ static Cipher::CipherAlgorithm selectCipherAlgorithm() {
     cout << "\n" << _("Enter the number corresponding to your choice: ");
     char answer[10];
     char *res = fgets(answer, sizeof(answer), stdin);
-    int cipherNum = (res == nullptr ? 0 : atoi(answer));
+    int cipherNum = (res == nullptr ? 0 : (int)strtol(answer, nullptr, 10));
```

Comment:
Hmm, where is the improvement here? Still no check if we got a valid number.

Reply:
Thx for your review !
Goal is at least to remove warnings.
There is a check just after `strtol()`, so it's OK.
In `XmlReader.cpp`, I've added checks which were not done before.
Thx !